components/brave_wallet_ui/components/market-datatable/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { AssetPriceChange } from '../asset-price-change'
+import useOnScreen from '../../common/hooks/on-screen'
+import { CoinGeckoText } from '../desktop/views/portfolio/style'
+
+export interface MarketDataHeader extends Header {
+  id: MarketDataTableColumnTypes
+}
+
+export interface Props {
+  headers: MarketDataHeader[]
+  coinMarketData: BraveWallet.CoinMarket[]
+  moreDataAvailable: boolean
+  showEmptyState: boolean
+  onFetchMoreMarketData: () => void
+  onSort?: (column: MarketDataTableColumnTypes, newSortOrder: SortOrder) => void
+}
+
+export const MarketDataTable = (props: Props) => {
+  const { headers, coinMarketData, moreDataAvailable, showEmptyState, onFetchMoreMarketData, onSort } = props
+  const ref: any = React.useRef<HTMLDivElement>()
+  const onScreen = useOnScreen<HTMLDivElement>(ref, {
+    rootMargin: '0px',
+    threshold: 0
+  })
+
+  React.useEffect(() => {
+    if (onScreen && moreDataAvailable) {
+      onFetchMoreMarketData()
+    }
+  }, [onScreen, moreDataAvailable])
+
+  const renderCells = (coinMarkDataItem: BraveWallet.CoinMarket) => {
+    const {
+      name,
+      symbol,
+      image,
+      currentPrice,
+      priceChange24h,
+      priceChangePercentage24h,
+      marketCap,
+      marketCapRank,
+      totalVolume
+    } = coinMarkDataItem
+
+    const formattedPrice = formatFiatAmountWithCommasAndDecimals(currentPrice.toString(), 'USD')
+    const formattedPercentageChange = formatPricePercentageChange(priceChangePercentage24h, 2, true)
+    const formattedMarketCap = formatPriceWithAbbreviation(marketCap.toString(), 'USD', 1)
+    const formattedVolume = formatPriceWithAbbreviation(totalVolume.toString(), 'USD', 1)
+    const isDown = priceChange24h < 0
+
+    const cellsContent: React.ReactNode[] = [
+      <AssetsColumnWrapper>
+        {/* Hidden until wishlist feature is available on the backend */}
+        {/* <AssetsColumnItemSpacer>
+          <AssetWishlistStar active={true} />
+        </AssetsColumnItemSpacer> */}
+        <AssetsColumnItemSpacer>
+          <TextWrapper alignment="center">{marketCapRank}</TextWrapper>
+        </AssetsColumnItemSpacer>
+        <AssetNameAndIcon
+          assetName={name}
+          symbol={symbol}
+          assetLogo={image}
+        />
+      </AssetsColumnWrapper>,
+
+      // Price Column
+      <TextWrapper alignment="right">{formattedPrice}</TextWrapper>,
+
+      // Price Change Column
+      <TextWrapper alignment="right">
+        <AssetPriceChange
+          isDown={isDown}
+          priceChangePercentage={formattedPercentageChange}
+        />
+      </TextWrapper>,
+
+      // Market Cap Column
+      <TextWrapper alignment="right">{formattedMarketCap}</TextWrapper>,
+
+      // Volume Column
+      <TextWrapper alignment="right">{formattedVolume}</TextWrapper>
+
+      // Line Chart Column
+      // Commented out because priceHistory data is yet to be
+      // available from the backend
```

Comment:
add link to backend issue

Reply:
@josheleonard There is not backend issue tracking this at the moment. The decision was to ship without it, maybe in the future.